### PR TITLE
[ASP.NET Core] Remove SDK dependency

### PR DESCRIPTION
**Pull request by vishweshbankwar**
Towards #4809

Please provide a brief description of the changes here.

## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (nullable enabled, static analysis, etc.)
* [ ] Unit tests added/updated
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: test/OpenTelemetry.Instrumentation.AspNetCore.Tests/BasicTests.cs [hunk 1]
```diff
@@ -663,14 +660,9 @@ public async Task HttpRequestMethodIsSetAsPerSpec(string originalMethod, string
     {
         var exportedItems = new List<Activity>();
 
-        var configuration = new ConfigurationBuilder()
```
Comment (cijothomas) [inline test/OpenTelemetry.Instrumentation.AspNetCore.Tests/BasicTests.cs:666]: Is this related to this PR or something accidentally creeped into this ?

Comment (vishweshbankwar) [reply test/OpenTelemetry.Instrumentation.AspNetCore.Tests/BasicTests.cs:666]: its leftover clean up from https://github.com/open-telemetry/opentelemetry-dotnet/pull/5066

File: src/OpenTelemetry.Instrumentation.AspNetCore/OpenTelemetry.Instrumentation.AspNetCore.csproj [hunk 2]
```diff
@@ -15,25 +15,25 @@
...
-    <Compile Include="$(RepoRoot)\src\Shared\EnvironmentVariables\*.cs" Link="Includes\EnvironmentVariables\%(Filename).cs" Condition="'$(TargetFrameworkIdentifier)' != '.NETCoreApp'" />
-    <Compile Include="$(RepoRoot)\src\Shared\HttpSemanticConventionHelper.cs" Link="Includes\HttpSemanticConventionHelper.cs" />
-    <Compile Include="$(RepoRoot)\src\Shared\Options\*.cs" Link="Includes\Options\%(Filename).cs" />
     <Compile Include="$(RepoRoot)\src\Shared\Shims\NullableAttributes.cs" Link="Includes\Shims\NullableAttributes.cs" />
   </ItemGroup>
 
   <ItemGroup Condition="'$(RunningDotNetPack)' != 'true'">
-    <ProjectReference Include="$(RepoRoot)\src\OpenTelemetry\OpenTelemetry.csproj" />
+    <ProjectReference Include="$(RepoRoot)\src\OpenTelemetry.Api\OpenTelemetry.Api.csproj" />
+    <ProjectReference Include="$(RepoRoot)\src\OpenTelemetry.Api.ProviderBuilderExtensions\OpenTelemetry.Api.ProviderBuilderExtensions.csproj" />
```
Comment (utpilla) [inline src/OpenTelemetry.Instrumentation.AspNetCore/OpenTelemetry.Instrumentation.AspNetCore.csproj:23]: You shouldn't need a reference to both API and ProviderBuilderExtensions projects.

Comment (utpilla) [reply src/OpenTelemetry.Instrumentation.AspNetCore/OpenTelemetry.Instrumentation.AspNetCore.csproj:23]: Do we need a reference to ProviderBuilderExtensions?

Comment (vishweshbankwar) [reply src/OpenTelemetry.Instrumentation.AspNetCore/OpenTelemetry.Instrumentation.AspNetCore.csproj:22]: Yes, we use extensions from it such as this one https://github.com/open-telemetry/opentelemetry-dotnet/blob/f9c300b2849f7552756d6ca23a6aaa054a117dd2/src/OpenTelemetry.Api.ProviderBuilderExtensions/Trace/OpenTelemetryDependencyInjectionTracerProviderBuilderExtensions.cs#L137

File: src/OpenTelemetry.Instrumentation.AspNetCore/OpenTelemetry.Instrumentation.AspNetCore.csproj [hunk 3]
```diff
@@ -15,25 +15,23 @@
...
   <ItemGroup Condition="'$(RunningDotNetPack)' == 'true'">
-    <PackageReference Include="OpenTelemetry" />
+    <PackageReference Include="OpenTelemetry.Api.ProviderBuilderExtensions" />
   </ItemGroup>
 
   <ItemGroup Condition="'$(TargetFrameworkIdentifier)' != '.NETCoreApp'">
     <PackageReference Include="System.Text.Encodings.Web" />
     <PackageReference Include="Microsoft.AspNetCore.Http.Abstractions" />
     <PackageReference Include="Microsoft.AspNetCore.Http.Features" />
+    <PackageReference Include="Microsoft.Extensions.Options" />
```
Comment (utpilla) [inline src/OpenTelemetry.Instrumentation.AspNetCore/OpenTelemetry.Instrumentation.AspNetCore.csproj:34]: Why is no explicit package reference required for this on `net6.0` and `net8.0`?

Comment (vishweshbankwar) [reply src/OpenTelemetry.Instrumentation.AspNetCore/OpenTelemetry.Instrumentation.AspNetCore.csproj:34]: for net6.0/net8.0, this package is part of the framework itself so we dont need explicit reference.

**Review by CodeBlanch — APPROVED** [review]
LGTM

**Review by utpilla — APPROVED** [review]

**Pull request merged**